### PR TITLE
Create selection_sort.dart #2154

### LINKED ISSUE #2154: Selection Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [ ] Dart - @pooja-git11
- [x] Go
- [x] Java
- [x] JS
- [x] Kotlin
- [x] Php - @BhardwajShweta25
- [x] Python
- [x] Readme
- [x] Ruby
- [x] Rust - @raksha009
- [ ] TypeScript - @lakshyajit165

### PR DESCRIPTION
**Pull request by pooja-git11**
Fixes #2154

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
complete your PR template

**Comment by asha15** [comment]
please include the issue number next to `Fixes#`

File: Selection_Sort/selection_sort.dart #2154 [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            arr[min_index] = temp;
+        }
+    } 
+}
+
+void print_element(List<int> arr)
+{
+    for(int ele in arr)
+        print(ele);
+
```
Comment (asha15) [inline Selection_Sort/selection_sort.dart #2154]: remove this line

File: Selection_Sort/selection_sort.dart #2154 [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        var ele = int.parse( stdin.readLineSync());
+        array.add(ele);
+    }
+   
+    selection_sort(array ,size);
+     
+    print("Sorted array is ");
+    print_element(array);
+}
```
Comment (asha15) [inline Selection_Sort/selection_sort.dart #2154]: keep a line space after this

File: Selection_Sort/selection_sort.dart #2154 [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+/*
+Enter the size of array 
+5
+Enter array element
+50 
+40 
+30 
+20 
+10
```
Comment (asha15) [inline Selection_Sort/selection_sort.dart #2154]: take this into one line

File: Selection_Sort/selection_sort.dart #2154 [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+Sorted array is
+10
+20
+30
+40
+50
```
Comment (asha15) [inline Selection_Sort/selection_sort.dart #2154]: take this into one line

**Comment by MastersAbh** [comment]
Hi @pooja-git11 are you working on this? 

Please update the issue number this PR is for and also give space after for, if etc in the code.

**Comment by pooja-git11** [comment]
Yes, I am working on it, I will do all changes that you asked.

File: Selection_Sort/selection_sort.dart #2154 [hunk 5]
```diff
@@ -0,0 +1,64 @@
+//selection sort
```
Comment (somya-kapoor) [inline Selection_Sort/selection_sort.dart #2154:1]: change the file name to Selection_Sort/Selection_Sort.dart

File: Selection_Sort/selection_sort.dart #2154 [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        print ( "Enter $i element " );
+        var ele = int.parse ( stdin.readLineSync());
+        array.add (ele);
+    }
+   
+    selection_sort (array ,size);
+     
+    print ("Sorted array is ");
+
```
Comment (somya-kapoor) [inline Selection_Sort/selection_sort.dart #2154:44]: Remove this line.

File: Selection_Sort/selection_sort.dart #2154 [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    List <int> array = List();
+    
+    for (int i = 0; i < size; i++)
+    {
+        print ( "Enter $i element " );
+        var ele = int.parse ( stdin.readLineSync());
+        array.add (ele);
+    }
+   
+    selection_sort (array ,size);
```
Comment (somya-kapoor) [inline Selection_Sort/selection_sort.dart #2154:41]: Remove the space before , and add space after ,

File: Selection_Sort/selection_sort.dart #2154 [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            temp = arr[i];
+            arr[i] = arr[min_index];
+            arr[min_index] = temp;
+        }
+    } 
+}
+
+main()
+{
+
```
Comment (somya-kapoor) [inline Selection_Sort/selection_sort.dart #2154:28]: remove this line

File: Selection_Sort/Selection_Sort.dart [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+main()
+{
+    print ( "Enter the size of array " );
+    int size = int.parse ( stdin.readLineSync() );
+    
+    List <int> array = List();
+    
+    for (int i = 0; i < size; i++)
+    {
+        print ( "Enter $i element " );
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:35]: Remove space after ( and before )

File: Selection_Sort/Selection_Sort.dart [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        var ele = int.parse ( stdin.readLineSync());
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:36]: Remove space after (

File: Selection_Sort/Selection_Sort.dart [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+            arr[i] = arr[min_index];
+            arr[min_index] = temp;
+        }
+    } 
+}
+
+main()
+{
+    print ( "Enter the size of array " );
+    int size = int.parse ( stdin.readLineSync() );
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:29]: Remove space after ( and before )

File: Selection_Sort/Selection_Sort.dart [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+            temp = arr[i];
+            arr[i] = arr[min_index];
+            arr[min_index] = temp;
+        }
+    } 
+}
+
+main()
+{
+    print ( "Enter the size of array " );
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:28]: Remove space after ( and before )

File: Selection_Sort/Selection_Sort.dart [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    
+    for (int i = 0 ; i  < n - 1 ; i++)
+    {
+        min_index = i;
+        
+        for ( int j = i + 1 ; j < n ; j++ )
+            if ( arr[j] < arr[min_index] )
+                 min_index = j;
+            
+        if ( i != min_index )
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:17]: Remove space after ( and before )

File: Selection_Sort/Selection_Sort.dart [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+void selection_sort ( List<int> arr, int n)
+{
+    int temp, min_index;
+    
+    for (int i = 0 ; i  < n - 1 ; i++)
+    {
+        min_index = i;
+        
+        for ( int j = i + 1 ; j < n ; j++ )
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:13]: Remove space after ( and before )

File: Selection_Sort/Selection_Sort.dart [hunk 15]
```diff
@@ -0,0 +1,61 @@
+//selection sort 
+
+import 'dart:io';
+
+void selection_sort ( List<int> arr, int n)
+{
+    int temp, min_index;
+    
+    for (int i = 0 ; i  < n - 1 ; i++)
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:9]: Remove space after ( and before )

File: Selection_Sort/Selection_Sort.dart [hunk 16]
```diff
@@ -0,0 +1,61 @@
+//selection sort 
+
+import 'dart:io';
+
+void selection_sort ( List<int> arr, int n)
```
Comment (somya-kapoor) [inline Selection_Sort/Selection_Sort.dart:5]: Remove space after ( .

**Comment by somya-kapoor** [comment]
Please squash the commits.

**Comment by pooja-git11** [comment]
@somya-kapoor i don't know how to squash the commit , if you can help me any way then you are welcome

**Comment by singh-shreya6** [comment]
If you are unable to squash, please open a new PR, and close this one. Else it won't be counted. Please do it by tomorrow.

**Comment by pooja-git11** [comment]
okay , i will open new PR

**Pull request closed without merging**